src/rust_codegen.rs

```diff
@@ -198,15 +198,37 @@ impl Visitor for CodegenVisitor {
         let instance_name =
             format_ident!("{}Instance", reg.id.name.to_case(Case::Pascal));
 
+        // 8 => proc_macro2::Literal::u8_suffixed(v.value as u8),
+        // 16 => proc_macro2::Literal::u16_suffixed(v.value as u16),
+        // 32 => proc_macro2::Literal::u32_suffixed(v.value as u32),
+        // 64 => proc_macro2::Literal::u64_suffixed(v.value as u64),
+        // 128 => proc_macro2::Literal::u128_suffixed(v.value as u128),
         let doc = reg.doc.join("\n");
         let instance_doc = format!("Instance of a [`{}`]", reg.id.name);
-        let conversion = match reg.width.value {
-            8 | 16 | 32 | 64 => quote! {
-                BitSet::<#width>::from(value)
-            },
-            _ => quote! {
-                BitSet::<#width>::try_from(value).unwrap()
-            },
+        let aligned = matches!(reg.width.value, 8 | 16 | 32 | 64 | 128);
+        let reset = match &reg.reset_value {
+            None => quote! { self.0 = BitSet::<#width>::ZERO },
+            Some(v) if aligned => {
+                let val = match reg.width.value {
+                    8 => proc_macro2::Literal::u8_suffixed(v.value as u8),
+                    16 => proc_macro2::Literal::u16_suffixed(v.value as u16),
+                    32 => proc_macro2::Literal::u32_suffixed(v.value as u32),
+                    64 => proc_macro2::Literal::u64_suffixed(v.value as u64),
+                    128 => proc_macro2::Literal::u128_suffixed(v.value),
+                    _ => panic!("can't happen"),
+                };
+                quote! { self.0 = BitSet::<#width>::from(#val) }
+            }
+            Some(v) => {
+                let val = proc_macro2::Literal::u128_suffixed(v.value);
+                quote! { self.0 = BitSet::<#width>::try_from(#val).unwrap()}
```

Comment:
Since the reset value is an integer literal, it seems like we can throw an error at compile time if its too big for the register, and then sort out infallible methods to use during codegen and avoid `unwrap`? It seem like that should be possible with the `small_int_large_bitset` you added in the `bitset` crate recently.

Reply:
I tried to add a "does the reset value fit into the register" check in the parser, but it didn't fit there since that's a semantic check rather than syntactic.  It doesn't fit in the code generator, since that doesn't allow the components to return errors.  I could add a check, but it would require changing the whole flow of the code generator, or just panicing.

Reply:
Ok. I would definitely trade a compile time `panic` or [`compile_error`](https://doc.rust-lang.org/std/macro.compile_error.html) for a runtime `unwrap`. But we can circle back and look at adding a semantic checker between parsing and codegen.

Reply:
Added a follow up issue 

- https://github.com/oxidecomputer/rsf/issues/10